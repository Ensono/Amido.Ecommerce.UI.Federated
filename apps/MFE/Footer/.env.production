ASSETS_PATH=//localhost:3003
GENERATE_SOURCEMAP=false
PORT=3003
REMOTE_URLS={ "mfe_header":"http://localhost:3001" }
LOG_LEVEL=warn
