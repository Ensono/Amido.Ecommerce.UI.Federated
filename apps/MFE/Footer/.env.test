ASSETS_PATH=//localhost:3003
GENERATE_SOURCEMAP=true
PORT=3003
REMOTE_URLS={ "mfe_text_positioner": "http://localhost:3004" }
LOG_LEVEL=info