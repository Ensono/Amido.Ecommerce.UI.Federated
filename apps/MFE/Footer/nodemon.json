{
  "ext": "ts,tsx,css",
  "ignore": ["build/*", "src/remote-entry/*"]
}
