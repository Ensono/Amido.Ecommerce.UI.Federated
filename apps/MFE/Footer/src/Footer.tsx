import React from 'react'

import {context, federateComponent} from '@batman/federate-component'
import {getRemoteUrls} from '@batman/remote-urls'

export {context}

const remotesUrls = getRemoteUrls()

const TextPositioner = federateComponent('mfe_text_positioner', './text-positioner', remotesUrls.mfe_text_positioner)

const Footer: React.FC = ({children}) => {
  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
    <footer data-testid="moodule-federated-footer" onClick={() => alert('ciao mamma footer')}>
      <strong>I AM THE FOOTER</strong>
      <TextPositioner loadingFallback="loading!" errorFallback="oh no!">
        {children}
      </TextPositioner>
    </footer>
  )
}

export default Footer
