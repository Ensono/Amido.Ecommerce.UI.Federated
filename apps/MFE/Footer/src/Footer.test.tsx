import {render, waitFor} from '@testing-library/react'

import Footer from './Footer'

describe('Footer', () => {
  it('renders as expected', async () => {
    const {container} = render(<Footer>foo bar</Footer>)
    await waitFor(() => {
      expect(container).toMatchSnapshot()
    })
  })
})
