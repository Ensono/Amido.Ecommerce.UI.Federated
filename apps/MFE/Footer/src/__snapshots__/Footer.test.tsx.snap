// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders as expected 1`] = `
<div>
  <footer
    data-testid="moodule-federated-footer"
  >
    <strong>
      I AM THE FOOTER
    </strong>
    loading!
  </footer>
</div>
`;
